content/en/blog/2024/prom-and-otel/index.md

```diff
@@ -0,0 +1,671 @@
+It also gives namespace access to the PodMonitor and ServiceMonitor.
+
+```yaml
+- apiGroups:
+    - monitoring.coreos.com
+  resources:
+    - servicemonitors
+    - podmonitors
+  verbs:
+    - '*'
+- apiGroups: ['']
+  resources:
+    - namespaces
+  verbs: ['get', 'list', 'watch']
+```
+
+## Additional OTel components for Kubernetes
+
+This section covers additional OTel Collector components you can use to capture
+Kubernetes metrics.
+
+Receiving data:
+
+- [Kubernetes Cluster Receiver](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/receiver/k8sclusterreceiver):
+  collects cluster-level metrics and entity events from the
+  [Kubernetes API server](https://kubernetes.io/docs/reference/command-line-tools-reference/kube-apiserver/)
+- [Kubernetes Objects Receiver](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/receiver/k8sobjectsreceiver):
+  collects (pull/watch) objects from the Kubernetes API server
+- [Kubelet Stats Receiver](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/receiver/kubeletstatsreceiver):
+  pulls node, pod, container, and volume metrics from the API server on a
+  [Kubelet](https://kubernetes.io/docs/reference/command-line-tools-reference/kubelet/)
+  and sends it down the metric pipeline for further processing
+- [Host Metrics Receiver](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/receiver/hostmetricsreceiver):
+  scrapes system metrics from hosts that make up the cluster
+
+Processing data:
+
+- [Kubernetes Attributes Processor](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/processor/k8sattributesprocessor):
+  adds Kubernetes context, thereby
+- enabling you to correlate application telemetry with your Kubernetes telemetry
+- – considered one of the most important components for monitoring Kubernetes
+- with OpenTelemetry
+
+You can also use the Kubernetes attributes processor to set custom resource
+attributes for traces, metrics, and logs using the Kubernetes labels and
+annotations you’ve added to your pods and namespaces.
+
+There are a few more Collector components you can implement to monitor
+Kubernetes, including Kubernetes-specific ones as well as general-use
+processors, such as the batch, memory limiter, and resource processors. To learn
+more, see
+[Important Components for Kubernetes](/docs/kubernetes/collector/components/).
+
+After you’ve configured the components in your Collector config file, you need
+to enable them within the [pipelines](/docs/collector/configuration/#pipelines)
+section. A data pipeline enables you to
+[collect](/docs/collector/configuration/#receivers),
+[process](/docs/collector/configuration/#processors), and route data from any
+source [to one destination or more](/docs/collector/configuration/#exporters).
+
+## Pros and cons
+
+The following are pros and cons of the setup we covered in this article.
```

Comment:
Is this compared to running the standard kube-prometheus stack?  (https://github.com/prometheus-operator/kube-prometheus). Or what is this being compared to?

Reply:
Pros and cons of having an "OTel native" approach to ingesting Prometheus metrics.

Reply:
@dashpole Based on ^^, do you have any suggestions for things that need to be edited?

Reply:
For example, the pro of "not having to maintain prometheus" applies to users that are running prometheus as their data store. It seemed like there was some alternative to the approach above you had in mind, but wasn't called out.

Maybe replace items related to maintaining prometheus, or using promQL with statements about flexibility?  E.g. Pro: Export data to any backend which has an collector exporter, or supports OTLP.